### DISCUSSION TITLE
New configuration for float window/popup

### DISCUSSION THREAD

**Discussion by chemzqm** [neoclide/coc.nvim#3351] [Show and tell]
Added configurations: `suggest.floatConfig`, `diagnostic.floatConfig`， `signature.floatConfig` and `hover.floatConfig`,
following properties are supported:

- `border`: Change to `true` to enable border.
- `highlight`: Background highlight group of float window.
- `title`: Title used by float window.
- `borderhighlight`: Border highlight group of float window.
- `close`: Set to true to draw close icon.
- `maxWidth`: Maximum width of float window, contains border.
- `maxHeight`: Maximum height of float window, contains border.
- `winblend`: Set 'winblend' option of window, neovim only.
- `focusable`: Set to false to make window not focusable, neovim only.
- `shadow`: Set to true to enable shadow, neovim only.

To make sure it's available, run `:h coc-config-float`.

Following configurations was **removed**:

- `hover.floatMaxWidth`
- `hover.floatMaxHeight`
- `suggest.maxPreviewWidth`
- `diagnostic.maxWindowHeight`
- `diagnostic.maxWindowWidth`
- `signature.maxWindowWidth`
- `signature.maxWindowHeight`

Start from commit eb47e40c

**Comment by kevinhwang91** [comment]
Now, coc looks more modern to me.
Have you forgotten the `signature.floatConfig`?
I add border and change the highlight for `diagnostic` and `hover`. BTW, as a tmux user, I want an `focusable` option :)

edited:  has supported `signature.floatConfig` and `focusable`

**Comment by kevinhwang91** [comment]
![2021-09-05_09-00](https://user-images.githubusercontent.com/17562139/132111572-8114f468-79fc-459d-aa89-ea9cf6892bfe.png)
look weird if I change the `borderhighlight`.

**Reply by chemzqm** [reply]
LGTM

**Reply by UltiRequiem** [reply]
LGTM

**Comment by kevinhwang91** [comment]
Found an issue for scrollbar in neovim:
![2021-09-05_20-46](https://user-images.githubusercontent.com/17562139/132127334-a67b7434-359a-48fd-a3d8-4652a4686d69.png)

**Reply by chemzqm** [reply]
What's wrong with it?

**Reply by kevinhwang91** [reply]
The lower part of the scrollbar is `  `, should it be padded with `|`?

**Reply by chemzqm** [reply]
No, neovim doesn't show float window content with lower index when using winblend, and I don't want to add text to scrollbar buffer

**Reply by kevinhwang91** [reply]
How about composing the scrollbar and the border as a single window?

**Reply by chemzqm** [reply]
Not worth it.

**Comment by oblitum** [comment]
When there's border in docs float, it gives the impression that the docs float is smaller/bellow than the pum menu (which can't have a border at all sadly) because the top border of the doc float occupies line height, with actual border line lying at the middle, hence it's unaligned with pum whose background covers all the line contents as a whole, giving the impression of an outer edge.

I don't have much idea whether that could be made better.

**Comment by Arghnews** [comment]
**edit** Works on neovim 0.4.4 so using that for now, will open an issue when I look into it further.

My floating windows used to work, now no longer. Using coc.nvim + nvim 0.5.1 + clangd

When I run `:h coc-config-float` I get 
`E149: Sorry, no help for coc-config-float`

Could you explain what this means? And also it would be great: what should I do about it?

Same behaviour with nvim 0.5.0 (stable) and 0.5.1.

Interestingly sometimes I do see the floating window as it flickers up then shuts.

Sorry if this isn't the appropriate place to post this, I'll submit an issue if necessary instead.

**Comment by oblitum** [comment]
@Arghnews I run 0.5.1 and have no issues, either with functionality as with the help entry.
